### ISSUE TITLE
ModuleNotFoundError (qpageview) on Fedora 38 KDE spin

### ISSUE THREAD

**Issue by RedBearAK** [Nyre221/dolphin-quick-view#3]
Not sure where else to go with this myself. 

What happens in the journal when I open a Dolphin window: 

```sh
Sep 10 16:07:22 fedora systemd[225583]: Started app-org.kde.dolphin-280b186f483449d7894f9410c83c0f6c.scope - Dolphin - File Manager.
Sep 10 16:07:22 fedora dolphin[228700]: kf.config.core: Access to ' "/home/testuser/Downloads/dolphin_quick_view/quick_view.desktop" ' denied, not owned by root and executable flag not set.
```

What happens when I try to Quick View a file: 

```sh
Sep 10 16:13:59 fedora dolphin[228700]: Cannot initialize model with data QJsonObject(). missing: QJsonValue(string, "urls")
Sep 10 16:14:13 fedora systemd[225583]: Started app-quick_view-1b9ab765583c42a29a4a67d607c0a91d.scope - Quick view.
Sep 10 16:14:13 fedora python3[230910]: detected unhandled Python exception in '/home/testuser/.config/quick_view/quick_view.pyz'
Sep 10 16:14:13 fedora plasmashell[230910]: Traceback (most recent call last):
Sep 10 16:14:13 fedora plasmashell[230910]:   File "<frozen runpy>", line 198, in _run_module_as_main
Sep 10 16:14:13 fedora plasmashell[230910]:   File "<frozen runpy>", line 88, in _run_code
Sep 10 16:14:13 fedora plasmashell[230910]:   File "/home/testuser/.config/quick_view/quick_view.pyz/__main__.py", line 1, in <module>
Sep 10 16:14:13 fedora plasmashell[230910]:   File "/home/testuser/.config/quick_view/quick_view.pyz/quick_view.py", line 8, in <module>
Sep 10 16:14:13 fedora plasmashell[230910]:   File "/home/testuser/.config/quick_view/quick_view.pyz/Page_viewer.py", line 1, in <module>
Sep 10 16:14:13 fedora plasmashell[230910]: ModuleNotFoundError: No module named 'qpageview'
```

Problem trying to install `qpageview` with `pip`: 

```sh
pip install qpageview PyQt5
Defaulting to user installation because normal site-packages is not writeable
Collecting qpageview
  Using cached qpageview-0.6.2.tar.gz (162 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [24 lines of output]
      Traceback (most recent call last):
        File "/usr/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/usr/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/usr/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-1kl9jjpa/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 355, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-1kl9jjpa/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 325, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-1kl9jjpa/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 507, in run_setup
          super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-1kl9jjpa/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 341, in run_setup
          exec(code, locals())
        File "<string>", line 32, in <module>
        File "/tmp/pip-install-6afp_seo/qpageview_26a27545f6b24413a6e729d07f96fb80/qpageview/__init__.py", line 65, in <module>
          from . import link
        File "/tmp/pip-install-6afp_seo/qpageview_26a27545f6b24413a6e729d07f96fb80/qpageview/link.py", line 34, in <module>
          from PyQt5.QtCore import pyqtSignal, QEvent, QRectF, Qt
      ModuleNotFoundError: No module named 'PyQt5'
```

And PyQt5 is already installed: 

```sh
pip install PyQt5
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: PyQt5 in /usr/lib64/python3.12/site-packages (5.15.9)
Requirement already satisfied: PyQt5-sip<13,>=12.11 in /usr/lib64/python3.12/site-packages (from PyQt5) (12.12.2)
DEPRECATION: textract 1.6.5 has a non-standard dependency specifier extract-msg<=0.29.*. pip 23.3 will enforce this behaviour change. A possible replacement is to upgrade to a newer version of textract or contact the author to suggest that they release a version with a conforming dependency specifiers. Discussion can be found at https://github.com/pypa/pip/issues/12063
```

**Comment by shawnk-23** [comment]
> ```shell
> ModuleNotFoundError: No module named 'qpageview'
> ```

I confirm that I get the same error on garuda: 'ModuleNotFoundError: No module named 'qpageview''

**Comment by RedBearAK** [comment]
Got past it on Fedora 38 by installing: 

```sh
sudo dnf install python3-qpageview
```

**Comment by shawnk-23** [comment]
> Got past it on Fedora 38 by installing:
> 
> ```shell
> sudo dnf install python3-qpageview
> ```

I guess Fedora doesn't have qpageview either. yeah that was the name I found from garuda package manager too. Is yours working now?

**Comment by RedBearAK** [comment]
@shawnk-23 

>  Is yours working now?

It is. No more errors in the journal. And I also grabbed the 1.1.1 release that has the shortcuts closer to what I was expecting. 

Disable the Dolphin shortcut for selecting files with Space if you want to close the preview window with Space, or there may be a weird loop in Dolphin.

**Comment by shawnk-23** [comment]
> @shawnk-23
> 
> > Is yours working now?
> 
> It is. No more errors in the journal. And I also grabbed the 1.1.1 release that has the shortcuts closer to what I was expecting.
> 
> Disable the Dolphin shortcut for selecting files with Space if you want to close the preview window with Space, or there may be a weird loop in Dolphin.

I finally got this working but hasn't been able to setup shortcut to work. how do I disalbe dolphin shortcut for selecting files with space? when i checked the dolphin shortcuts nothing was assigned to spacebar. I'm trying to configure it the same way as gnome sushi or macbook quick view. basically open/close with spacebar. how can i set it up like that? any idea?

**Comment by shawnk-23** [comment]
> @shawnk-23
> 
> > Is yours working now?
> 
> It is. No more errors in the journal. And I also grabbed the 1.1.1 release that has the shortcuts closer to what I was expecting.
> 
> Disable the Dolphin shortcut for selecting files with Space if you want to close the preview window with Space, or there may be a weird loop in Dolphin.

btw, is your pdf/mp4 preview working? mine doesn't. i have all required packages installed. i wonder why...

**Comment by RedBearAK** [comment]
@shawnk-23 

They just call it "Space" in the Dolphin shortcuts dialog. It should show up when you go in the "hamburger" menu and go in the Configure submenu, "Configure Keyboard Shorcuts...". Search for either "space" or "select". Set it to "Custom" and click the delete button to leave it set to nothing. 

I haven't gotten to the part where I try to set it up to open with Space. I got distracted. But you can now at least close it with Space. 

I'm not sure if it will be possible to assign Space to activate the new menu item, or to activate the script only while in Dolphin. I think Dolphin may need to be patched to allow that to be possible. 

There are these instructions on the main page, but I think that just makes a global shortcut, and making a global shortcut assigned to Space would be impractical. 

> 4: If you want to open it with a shortcut and not just via dolphin's context menu: go to plasma settings -> shortcuts -> add a shortcut that executes a command: /home/YOUR_USERNAME/.config/quick_view/dolphin_quick_view_shortcut.sh

**Comment by RedBearAK** [comment]
@shawnk-23 

> btw, is your pdf/mp4 preview working? mine doesn't. i have all required packages installed. i wonder why...

I don't know, I've just been trying to test this in a couple of VMs with Fedora and ArcoLinux. I don't really have files like that in the VMs yet. 

I'm sure this project will get much better. The repo has only been active for less than a day at this point. Have patience.

**Comment by shawnk-23** [comment]
> @shawnk-23
> 
> They just call it "Space" in the Dolphin shortcuts dialog. It should show up when you go in the "hamburger" menu and go in the Configure submenu, "Configure Keyboard Shorcuts...". Search for either "space" or "select". Set it to "Custom" and click the delete button to leave it set to nothing.
> 
> I haven't gotten to the part where I try to set it up to open with Space. I got distracted. But you can now at least close it with Space.
> 
> I'm not sure if it will be possible to assign Space to activate the new menu item, or to activate the script only while in Dolphin. I think Dolphin may need to be patched to allow that to be possible.
> 
> There are these instructions on the main page, but I think that just makes a global shortcut, and making a global shortcut assigned to Space would be impractical.
> 
> > 4: If you want to open it with a shortcut and not just via dolphin's context menu: go to plasma settings -> shortcuts -> add a shortcut that executes a command: /home/YOUR_USERNAME/.config/quick_view/dolphin_quick_view_shortcut.sh

Thanks for the tips. that's where I checked already but it was already empty. So far, what worked for me is ctrl+space to open and q to close. For the time being it does the job. I'm happy with it for now. :)

**Comment by Nyre221** [comment]
@RedBearAK  @shawnk-23 
> There are these instructions on the main page, but I think that just makes a global shortcut, and making a global shortcut assigned to Space would be impractical.

The script does not start quick view if the dolphin window is not active.

> I'm not sure if it will be possible to assign Space to activate the new menu item, or to activate the script only while in Dolphin. I think Dolphin may need to be patched to allow that to be possible.

The problem with using a global shortcut is that it only accepts key combinations, not a single key.

to fix this you can use some other program to add the shortcut, but it's probably not an ideal solution. You would probably have problems creating files with spaces in the name.

I close this report. the original problem was due to incorrect writing of the requirements.

**Comment by RedBearAK** [comment]
@Nyre221 

> The script does not start quick view if the dolphin window is not active.

I noticed that. I was able to assign a long multi-key global shortcut and it wouldn't activate the Quick View window unless I was in Dolphin. Very clever. 

> You would probably have problems creating files with spaces in the name.

Yes, this is the central issue, and I think that's why Dolphin will need to be patched somehow so that Space can be used to activate something like Quick View while not in some kind of text field. I never thought about it, but it works fine in Nautilus without interfering with the ability to rename things. Must be inherent to Nautilus or activated by the Sushi plugin. 

I have a complicated "fix" in my keymapper setup for using the Enter key to rename files in Linux file managers the same way it works in the Finder in macOS. It toggles back and forth between Enter being "F2" and "Enter". Works, but has edge cases where you get unexpected results. Something similar would have to be done to allow using Space to activate Quick View right now. Sad, but that's what we have to work with until they fix Dolphin. 

Unfortunately I'm using most of the possible Mod+Space combos already for other things.
